### DISCUSSION TITLE
How to create an WebHDFS connection?

### DISCUSSION THREAD

**Discussion by ly-dlau** [apache/airflow#35948] [General]
Hi, I'm using airflow 2.7.3 with HDFS provider 4.2.0. 
I can't create an WebHDFS / HDFS connection because the hdfs hook is no longer available.
In addition, in the Add Connection page, under Connection Type, I don't see any `HDFS` / `WebHDFS` options available.
I need them for the Livy operator. I need to upload a python script to HDFS first before I execute it using livy.

I do have the provider installed:
<img width="1058" alt="image" src="https://github.com/apache/airflow/assets/152390344/d9c84807-362b-42dc-b364-79177f97739c">

To double check, I made sure pip has it installed
```
$ pip freeze | grep hdfs
apache-airflow-providers-apache-hdfs==4.2.0
hdfs==2.7.3
```

Question: How shouldI create an hdfs connection ?

Cheers

**Comment by tanvn** [comment]
@potiuk @eladkal 
Hi, I understand that this is quite old but the fact is I am also looking for a tutorial about how to create a WebHDFS connection.
Do you have any idea?

**Reply by eladkal** [reply]
You should look into the HDFS provider docs:
https://airflow.apache.org/docs/apache-airflow-providers-apache-hdfs/stable/connections.html

From the doc main page you can navigate to all the docs (airflow, helm chart, providers...)
https://airflow.apache.org/docs/

**Reply by tanvn** [reply]
Hmm, actually, I have checked the documents as well as the source code of https://github.com/apache/airflow/blob/2.10.5/airflow/providers/apache/hdfs/hooks/webhdfs.py
before posting this question.
I think it is better to have an example.
So, I tried creating a connection from Web UI as below and with Kerberos authentication (by setting `core.security = "kerberos"` configuration) it is working.
(Not sure what is the purpose of `schema` field, though)

<img width="1698" alt="Screenshot 2025-05-06 at 0 02 30" src="https://github.com/user-attachments/assets/11c9d08a-6540-446a-bbbe-5da140ca3cd6" />
